File: test/Data.Analyzers.Vsix.Test/VisualBasic/MissingForeignKeyImplementation.vb
Imports System.Data.SqlClient
Imports DevZest.Data.Annotations

Public Class MissingForeignKeyImplementation
    Inherits ForeignKeyDiagnosticsBase

    Protected Sub New(sqlConnection As SqlConnection)
        MyBase.New(sqlConnection)
    End Sub

    Private m_Addresses As DbTable(Of Address)
    Public ReadOnly Property Addresses As DbTable(Of Address)
        Get
            Return GetTable(m_Addresses)
        End Get
    End Property

    Private m_Customers As DbTable(Of Customer)
    <ForeignKey("FK_Customer_Address")>
    Public ReadOnly Property Customers As DbTable(Of Customer)
        Get
            Return GetTable(m_Customers)
        End Get
    End Property
End Class

File: test/Data.Analyzers.Vsix.Test/CSharp/MissingForeignKeyImplementation.cs
using DevZest.Data.Annotations;
using System.Data.SqlClient;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class MissingForeignKeyImplementation : ForeignKeyDiagnosticsBase
    {
        public MissingForeignKeyImplementation(SqlConnection sqlConnection)
            : base(sqlConnection)
        {
        }

        private DbTable<Address> _addresses;
        public DbTable<Address> Addresses
        {
            get { return GetTable(ref _addresses); }
        }

        private DbTable<Customer> _customers;
        [ForeignKey("FK_Customer_Address")]
        public DbTable<Customer> Customers
        {
            get { return GetTable(ref _customers); }
        }
    }
}
